CSC8503/NetworkPlayer.cpp

```diff
@@ -279,66 +279,69 @@ void NetworkPlayer::RayCastFromPlayer(GameWorld* world, float dt) {
 	if (isEquippedItemUsed) {
 		ItemUseType equippedItemUseType = mInventoryBuffSystemClassPtr->GetPlayerInventoryPtr()->GetItemUseType(GetEquippedItem());
 		if (equippedItemUseType != ItemUseType::NeedInteractableToUse)
-			return;
-		isRaycastTriggered = true;
+			return false;
 		interactType = NCL::CSC8503::InteractType::ItemUse;
+
+		return true;
 	}
+	return false;
+}
 
-	if (isRaycastTriggered) {
-		Ray ray;
-		if (!mIsLocalPlayer) {
-			ray = mPlayerInputs.rayFromPlayer;
-		}
-		else {
-			ray = CollisionDetection::BuidRayFromCenterOfTheCamera(world->GetMainCamera());
-		}
-		std::cout << "Ray fired" << std::endl;
-		RayCollision closestCollision;
 
-		if (world->Raycast(ray, closestCollision, true, this)) {
-			auto* objectHit = (GameObject*)closestCollision.node;
-			if (objectHit) {
-				Vector3 objPos = objectHit->GetTransform().GetPosition();
+void NetworkPlayer::RayCastFromPlayer(GameWorld* world, const NCL::CSC8503::InteractType& interactType, float dt) {
+	Ray ray;
+	if (!mIsLocalPlayer) {
+		ray = mPlayerInputs.rayFromPlayer;
+	}
+	else {
+		ray = CollisionDetection::BuidRayFromCenterOfTheCamera(world->GetMainCamera());
+	}
+	std::cout << "Ray fired" << std::endl;
+	RayCollision closestCollision;
+
+	if (world->Raycast(ray, closestCollision, true, this)) {
+		auto* objectHit = (GameObject*)closestCollision.node;
+		if (objectHit) {
+			Vector3 objPos = objectHit->GetTransform().GetPosition();
 
-				Vector3 playerPos = GetTransform().GetPosition();
+			Vector3 playerPos = GetTransform().GetPosition();
 
-				float distance = (objPos - playerPos).Length();
+			float distance = (objPos - playerPos).Length();
 
-				if (distance > 17.5f) {
-					std::cout << "Nothing hit in range" << std::endl;
-					return;
-				}
+			if (distance > 17.5f) {
+				std::cout << "Nothing hit in range" << std::endl;
+				return;
+			}
 
-				//Check if object is an item.
-				Item* item = dynamic_cast<Item*>(objectHit);
-				if (item != nullptr) {
-					item->OnPlayerInteract(mPlayerID);
-					return;
-				}
+			//Check if object is an item.
+			Item* item = dynamic_cast<Item*>(objectHit);
```

Comment:
We're avoiding dynamic_cast due to PS5 not liking it, find another way to cast this such as static_cast